js/index.js

```diff
@@ -17,3 +17,30 @@ document.querySelectorAll('.menu-link').forEach((link) => link.addEventListener(
   sideBar.classList.remove('active');
   wrapper.classList.remove('show');
 }));
+
+// preserve data
+const formContainer = document.querySelector('.contact_container');
+const emailValue = document.querySelector('.email');
+const messageError = document.querySelector('.errorMessage');
+
+formContainer.addEventListener('submit', (e) => {
+  if (emailValue.value === emailValue.value.toLowerCase()) {
+    messageError.textContent = '';
+    const formData = {
+      name: document.querySelector('name').value,
+      email: document.querySelector('email').value,
+      message: document.querySelector('message').value,
+    };
+    localStorage.setItem('formData', JSON.stringify(formData));
+  } else {
+    e.preventDefault();
+    messageError.innerHTML = '*Write your email in lower case <br> * form invalid';
+  }
+});
+
+if (formContainer) {
+  const { name, email, message } = JSON.parse(localStorage.getItem('formData'));
```

Comment:
- [ ] The data is not getting filled in the inputs on page reload. Kindly a load eventlistener on the window so that it fetches the data from the localStorage on load.
- [ ] Also please make sure the data is actually getting saved to the localstorage when the value of any input changes.

Reply:
Issue resolved